src/components/v2/devtronStackManager/DevtronStackManager.utils.ts

```diff
@@ -0,0 +1,148 @@
+    },
+    moreModules: {
+        id: 'moreModules',
+        name: 'More modules coming soon',
+        info: "You can also raise a request for a module that will improve your workflow.",
+        icon: 'moreModules',
+        installationStatus: ModuleStatus.NONE,
+    },
+    unknown: {
+        id: 'unknown',
+        name: 'New module coming soon',
+        info: "We're building a suite of modules to serve your software delivery lifecycle.",
+        icon: 'unknown',
+        installationStatus: ModuleStatus.NONE,
+    }
+}
+
+export const MODULE_DETAILS_INFO: Record<string, ModuleDetailsInfo> = {
+    cicd: {
+        name: 'Build and Deploy (CI/CD)',
+        infoList: [
+            'Continuous integration (CI) and continuous delivery (CD) embody a culture, set of operating principles, and collection of practices that enable application development teams to deliver code changes more frequently and reliably. The implementation is also known as the CI/CD pipeline.',
+            'CI/CD is one of the best practices for devops teams to implement. It is also an agile methodology best practice, as it enables software development teams to focus on meeting business requirements, code quality, and security because deployment steps are automated.',
+        ],
+        featuresList: [
+            "Discovery: What would the users be searching for when they're looking for a CI/CD offering?",
+            'Detail: The CI/CD offering should be given sufficient importance (on Website, Readme). (Eg. Expand capability with CI/CD module [Discover more modules])',
+            'Installation: Ability to install CI/CD module with the basic installation.',
+            'In-Product discovery: How easy it is to discover the CI/CD offering primarily once the user is in the product. (Should we talk about modules on the login page?)',
+        ],
+    },
+}
+
+export const ModulesSection: StackManagerNavLinkType[] = [
+    {
+        name: 'Discover',
+        href: URLS.STACK_MANAGER_DISCOVER_MODULES,
+        icon: DiscoverIcon,
+        className: 'discover-modules__nav-link',
+    },
+    {
+        name: 'Installed',
+        href: URLS.STACK_MANAGER_INSTALLED_MODULES,
+        icon: InstalledIcon,
+        className: 'installed-modules__nav-link',
+    },
+]
+export const AboutSection: StackManagerNavLinkType = {
+    name: 'About Devtron',
+    href: URLS.STACK_MANAGER_ABOUT,
+    icon: DevtronIcon,
+    className: 'about-devtron__nav-link',
+}
+
+const actionTriggered = (location: any, history: any) => {
+    const queryParams = new URLSearchParams(location.search)
+    queryParams.set('actionTriggered', 'true')
+    history.push(`${location.pathname}?${queryParams.toString()}`)
+}
+
+export const handleAction = async (
+    moduleName: string,
+    isUpgradeView: boolean,
+    upgradeVersion: string,
+    canUpdateServer: boolean,
+    setShowManagedByDialog: React.Dispatch<React.SetStateAction<boolean>>,
+    location: any,
```

Comment:
Please provide type of location and history.

Reply:
Sure, will try.

Reply:
Done